app/src/main/java/org/simple/clinic/signature/SignatureActivity.kt

```diff
@@ -1,20 +1,92 @@
 package org.simple.clinic.signature
 
+import android.content.Context
+import android.graphics.Bitmap
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import com.jakewharton.rxbinding3.view.clicks
+import io.reactivex.rxkotlin.ofType
 import kotlinx.android.synthetic.main.activity_signature.*
+import org.simple.clinic.ClinicApp
 import org.simple.clinic.R
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.util.unsafeLazy
+import javax.inject.Inject
 
 class SignatureActivity : AppCompatActivity(), SignatureUi {
+
+  private lateinit var component: SignatureComponent
+
+  private val context: Context = this
```

Comment:
We don't have to do this, since we are building the `MobiusDelegate` lazily we can directly get the context there

```kt
    MobiusDelegate.forActivity(
         events = events.ofType(),
         defaultModel = SignatureModel.create(filesDir),
         update = SignatureUpdate(),
         effectHandler = effectHandlerFactory.create(this).build()
     )
```